test/common/typesRepresentatives.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+var arrays = [
+	{
+		input: [],
+		description: 'empty array',
+		expectation: 'array'
+	},
+	{
+		input: ['abc'],
+		description: 'non empty array',
+		expectation: 'array'
+	}
+];
+
+var booleans = [
+	{
+		input: true,
+		description: 'true',
+		expectation: 'boolean'
+	},
+	{
+		input: false,
+		description: 'false',
+		expectation: 'boolean'
+	}
+];
+
+var integers = [
```

Comment:
there are many cases where we only accept positive numbers, so it be useful if we subdivide integers into positive integers and negative integers.

Reply:
nice

Reply:
actuallt isn't it connected with min / max?